src/ccs/r1cs.rs

```diff
@@ -121,4 +170,48 @@ pub mod tests {
         r1cs.check_relation(&z).unwrap();
         r1cs.relax().check_relation(&z).unwrap();
     }
+
+    // TestCircuit implements the R1CS for: x^3 + x + 5 = y (example from article
+    // https://www.vitalik.ca/general/2016/12/10/qap.html )
+    #[derive(Clone, Copy, Debug)]
+    pub struct TestCircuit<F: PrimeField> {
+        pub x: F,
+        pub y: F,
+    }
```

Comment:
Why do we have this struct under `r1cs.rs` instead of some testing file? Under `test` feature ofc.

Reply:
It is under the `pub mod tests` under the `#[cfg(test)]` feature. I placed it in the `r1cs.rs` file because it's a test circuit in R1CS.
Do you think that it would make more sense to move it into another place?

Reply:
I think if we have a good frontend trait, this would not need to be under `r1cs.rs` but rather in a more general module where it can also be used for CCS.

But in any case, if it's under test feat *which I missed*, it's ok

Reply:
You are right! Sorry for the mess.
I was checking this again, this `TestCircuit` is the 'handmade' circuit in `src/ccs/r1cs.rs`, while there is the `CubicFCircuit` and `CustomFCircuit` already existing in `src/frontend/mod.rs`.

So I agree that the one in `src/ccs/r1cs.rs` should not be there.

What I just did is I've removed the whole `TestCircuit` from `src/ccs/r1cs.rs` and in the only place that it was used (in `src/frontend/nova/decider.rs`) I've replaced it by the usage of the `WrapperCircuit` with the `CubicFCircuit` inside. This was what I already did for another test in that file, but forgot to do it for this one. So now the `src/ccs/r1cs.rs::tests::TestCircuit` no longer exists, and the test circuits are available in `src/frontend/mod.rs::tests`.
Thanks for pointing this out!